processing/src/main/java/org/apache/druid/segment/nested/NestedFieldLiteralColumnIndexSupplier.java

```diff
@@ -124,6 +131,10 @@ public <T> T as(Class<T> clazz)
       return (T) new NestedLiteralDictionaryEncodedStringValueIndex();
     }
 
+    if (skipPredicateIndex && clazz.equals(DruidPredicateIndex.class)) {
+      return null;
+    }
```

Comment:
Technically, this would indicate that this column cannot implement the given the interface, so the filter in question should fall back to another interface (i.e. it might make the filter to cascade down into trying an even worse strategy).  Instead, we should return a `DruidPredicateIndex` that returns `null` when asked to compute the index, indicating that it was unable to compute the index.  This will make it easier to do this "correctly" when we make the change to have this object able to also return ValueMatcher objects.